pkg/util/tar_utils.go

```diff
@@ -58,17 +58,22 @@ func unpackTar(tr *tar.Reader, path string) error {
 
 		// if its a dir and it doesn't exist create it
 		case tar.TypeDir:
-			if _, err := os.Stat(target); err != nil {
+			if _, err := os.Stat(target); os.IsNotExist(err) {
 				if err := os.MkdirAll(target, mode); err != nil {
-					glog.Errorf("Error creating directory %s while untarring", target)
 					return err
 				}
 				continue
 			}
 
 		// if it's a file create it
 		case tar.TypeReg:
-
+			// It's possible for a file to be included before the directory it's in is created.
+			baseDir := filepath.Dir(target)
+			if _, err := os.Stat(baseDir); os.IsNotExist(err) {
+				if err := os.MkdirAll(baseDir, 0755); err != nil {
```

Comment:
should we use the inherited `mode` from the header? `header.FileInfo().mode()` above

Reply:
Unfortunately the file mode doesn't work for the directory mode. I'm not really sure what the right thing to do here is.

Reply:
Ah yeah I see, I'm not really sure either. Maybe we create the directory with these permissions and then create the files inside of it with the header's permissions? I'm just worried that file permissions will be different than what's expected.

Reply:
Yeah how's this?